clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerClientInitialResponse.java

```diff
@@ -97,21 +135,41 @@ public SaslExtensions extensions() {
         return message.getBytes(StandardCharsets.UTF_8);
     }
 
+    /**
+     * Return the always non-null token value
+     * 
+     * @return the always non-null toklen value
+     */
     public String tokenValue() {
         return tokenValue;
     }
 
+    /**
+     * Return the always non-null authorization ID
+     * 
+     * @return the always non-null authorization ID
+     */
     public String authorizationId() {
         return authorizationId;
     }
 
     /**
      * Validates that the given extensions conform to the standard. They should also not contain the reserve key name {@link OAuthBearerClientInitialResponse#AUTH_KEY}
      *
+     * @param extensions
+     *            optional extensions to validate
+     * @return always non-null (potentially empty) extensions
```

Comment:
Update javadoc since there is no return value?

Reply:
👍

Reply:
@rajinisivaram @stanislavkozlovski I like the idea of SaslExtensionsCallback starting out with empty extensions so that if it isn't handled the caller will not receive null; I think this is a reasonable approach for dealing with the backwards compatibility issue. Should I make the change as part of this PR?  If so, I think I would add `public static final SaslExtensions NO_SASL_EXTENSIONS = ...` to the SaslExtensions class and refer to that in both the `SaslExtensionsCallback` and `OAuthBearerClientInitialResponse` constructors.  (I'll also fix "Update javadoc since there is no return value")

Reply:
@rondagostino Thank you, sounds good.